common/openpype_common/distribution/addon_distribution.py

```diff
@@ -0,0 +1,238 @@
+import os
+from enum import Enum
+from abc import abstractmethod
+import attr
+import logging
+import requests
+import platform
+import shutil
+
+from common.openpype_common.distribution.file_handler import RemoteFileHandler
+
+
+class UrlType(Enum):
+    HTTP = "http"
+    GIT = "git"
+    FILESYSTEM = "filesystem"
+
+
+class UpdateState(Enum):
+    EXISTS = "exists"
+    UPDATED = "updated"
+    FAILED = "failed"
+
+
+@attr.s
+class MultiPlatformPath(object):
+    windows = attr.ib(default=None)
+    linux = attr.ib(default=None)
+    darwin = attr.ib(default=None)
+
+
+@attr.s
+class AddonSource(object):
```

Comment:
Comment which was not written, only discussed:
I would not make this strict and define required attributes. From my point of view only require attribute of addon source should be `"type"`. For each type should be dedicated type handler (what `AddonDownloader` is right now) which validate rest of the attributes.

I'm 99.99% convinced each type handler will require it's attributes with possible different types - and they won't be the same for all sources. Even now there is `url` and `path`. By adding other source type handlers the attribute list would grows and will be unmaintainable with each new source handler. Any possible future attribute change would cause clashes and custom implementations would be a lottery.

Reply:
Updated AddonSource, created 2 classes inheriting from it.